build/Dockerfile

```diff
@@ -234,8 +234,8 @@ RUN --mount=type=bind,target=/tmp mkdir -p /var/lib/nginx /etc/nginx/secrets /et
 	&& setcap -v 'cap_net_bind_service=+eip' /usr/sbin/nginx 'cap_net_bind_service=+eip' /usr/sbin/nginx-debug \
 	&& [ -z "${BUILD_OS##*plus*}" ] && PLUS=-plus; cp -a /tmp/internal/configs/version1/nginx$PLUS.ingress.tmpl /tmp/internal/configs/version1/nginx$PLUS.tmpl \
 	/tmp/internal/configs/version2/nginx$PLUS.virtualserver.tmpl /tmp/internal/configs/version2/nginx$PLUS.transportserver.tmpl / \
-	&& chown -R 101:0 /etc/nginx /var/cache/nginx /var/lib/nginx /*.tmpl \
-	&& chmod -R g=u /etc/nginx /var/cache/nginx /var/lib/nginx /*.tmpl \
+	&& chown -R 101:0 /etc/nginx /var/cache/nginx /var/lib/nginx /var/log/nginx /*.tmpl \
+	&& chmod -R g=u /etc/nginx /var/cache/nginx /var/lib/nginx /var/log/nginx /*.tmpl \
```

Comment:
@lucacome, back in a24f8372cc1f7d46e01e117d65e68c5ec8026d97 you changed the default logging targets to `stderr` and `stdout`. Is it preferred to make `/var/log/nginx` writeable for compatibility?

If so, I am also considering splitting this into a separate small PR, as it would be a no-op for current `runAsUser: 101` deployments. Then it would be more straight forward to test the OpenShift change scope by using `edge-ubi` image, instead having to build manually.

Reply:
The problem with writing logs to file is that there is no logrotate within a container, so we run into a disk filling problem.
Also, stderr and stdout means that K8s intercepts the messages and they are associated with the pod through kubectl.

We do have customers that write log to file for very specific 3rd party tool purposes, but it is rare and the logrotate solution is complicated.

Therefore as a rule stderr and stdout is the common assumption.

Reply:
I am just wondering whether it is considered a supported solution. Sounds like it is for those specific customers. Will open a PR in that case.

Reply:
NGINX supports any path for any files.  There just happen to be defaults.
Are you thinking of how this interacts with readOnlyRootFileSystem or just the default log path permissions in general?
I mean, there is no harm with setting to correct group membership for the rare case.

Reply:
r/o root filesystem has a mountpoint to support `/var/log/nginx` scenario. No issues expected there.

This PR is concerned additional constraints that OpenShift wants to impose by default. Notably, it wants to run with a randomized UID/GID, where GID 0 is assigned as supplemental for filesystem actions.

Will try and phrase it nicely in the separate PR. 🤞🏻

Reply:
Opened #4269 for that scope. Removing the log directory change from this PR.